src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -395,7 +396,16 @@ public String getActiveDeadlineSecondsStr() {
     }
 
     public Map<String, String> getLabelsMap() {
-        return ImmutableMap.of("jenkins/label", label == null ? DEFAULT_LABEL : sanitizeLabel(label));
+        if(label == null) {
+            return ImmutableMap.of("jenkins/" + DEFAULT_LABEL, "true");
+        }
+
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        getLabelSet().stream().map(Objects::toString).sorted().forEach((atom) -> {
+            builder.put("jenkins/" + sanitizeLabel(atom), "true");
+        });
+
+        return builder.build();
```

Comment:
Would effectively revert [JENKINS-60088](https://issues.jenkins-ci.org/browse/JENKINS-60088)

Exposed labels in Kubernetes have no effect on the Jenkins labels. I would recommend reverting this hunk.

Reply:
It depends on the length of the labels. As sanitize reduces the size to the right most 63 characters, it is easier to create duplicate values. When the plugin queries for already running or pending pods during provisioning of new ones, this leads to pods not being created even if caps are not hit.

E.g. we create our labels based on capabilities which are installed in the docker image the label specified in the pod template might be linux-amd64-some_name_of_some_cap1-some_name_of_some_cap2-some_name_of_some_cap3, when we now have linux-ppc64le-some_name_of_some_cap1-..., this will yield a label of "ame_of_some_cap1-some_name_of_some_cap2-some_name_of_some_cap3".

Reply:
As a possible prevention of key explosion for JENKINS-60088 maybe using "jenkins/label-1=sanitizedLabel1, jenkins/label-2=saniziedLabel2, ..."

Reply:
I'd rather avoid "dynamic" labels due to the ticket above. To prevent collisions, we could instead add a label whose value would be a digest of the complete label?

Reply:
Some digest sounds good to me. Will have a look.